infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/FsmTaskManager.java

```diff
@@ -241,18 +242,19 @@ private void processReplicationTask(LogReplicationEvent currEvent, LogReplicatio
                 }
             }
 
-            removeReplicationEventIdFromMap(session, currEvent.getEventId());
```

Comment:
not related to this PR but can the notifyAll() on L240 be removed?  I do not think any test depends on it.

Reply:
This is used by atleast logReplicationIT in the cleanups between the tests. There maybe other suites who are depending on this as well.